docs/src/pages/guides/new-notifications.md

```diff
@@ -0,0 +1,173 @@
+---
+id: new-notifications
+title: New Notifications
+---
+
+## New Notifications
+
+Out of the box, Monika support a multitude of notification channels, from chat applications such as Slack, Discourse, Microsoft Teams, Google Chats to email based platform such as Mailgun, to generic webhooks and the SMTP protocol ([click here for the full list](https://monika.hyperjump.tech/guides/notifications)). Should you have a favorite tool not already supported, we invite you to add it to Monika's growing list. Here's how.
+
+## Prerequisites
+
+Adding a custom notification to Monika is quite straight forward. There are just a few requirements.
+
+1. Some basci javascript may be helpful.
+2. Some familiarity with git is required to clone and create pull requests.
+3. A Github account to fork and clone the [Monika open source repository](https://github.com/hyperjumptech/monika).
+4. A development environment already set up for javascript/typescript. See the project [README](https://github.com/hyperjumptech/monika/blob/main/README.md) for an overview.
+5. A code editor such as vim or visual code is recommended.
+
+## Extending
+
+The first step in adding custom notification is to extend the existing BaseNotification.
+
+1. Extend your new interface in `src/interfaces/notification.ts`
+
+Inside `notification.ts` define a new type, something to identify your new app/channel externally. In the example below, the google chat app is defined as `"google-chat"` with a `GoogleChatData` data type.
+
+```typescript
+interface GoogleChatNotification extends BaseNotification {
+  type: 'google-chat'
+  data: GoogleChatData
+}
+```
+
+2. Add your app specific data to `src/interfaces/data.ts`
+
+Next define all your required data in `data.ts`. If your data is a bit complex, you can define them in a separate module such as `whatsapp.ts`. For our example, google chat only requires a url as input data.
```

Comment:
Most of the time, your notification channel may need some required data such as API Key, Namespace Key, etc. You can define all your required data in `data.ts`.